### PR TITLE
#3939 & #3940: Wallet signing phrase and welcome screens

### PR DESCRIPTION
**Pull request by alwx**
Fixes two issues:

#3939: Introducing wallet / welcome screen
#3940: Introducing wallet / signing phrase

### CODE REVIEW

File: src/status_im/data_store/realm/schemas/base/v1/account.cljs [hunk 1]
```diff
@@ -2,23 +2,25 @@
...
-                          :signed-up?          {:type    :bool
-                                                :default false}
-                          :network             :string
-                          :networks            {:type       :list
-                                                :objectType :network}
-                          :settings            {:type :string}
-                          :sharing-usage-data? {:type :bool :default false}
-                          :dev-mode?           {:type :bool :default false}
-                          :seed-backed-up?     {:type :bool :default false}}})
+             :properties {:address               :string
```
Comment (flexsurfer) [inline src/status_im/data_store/realm/schemas/base/v1/account.cljs]: v1 in the release already

**Review by flexsurfer — CHANGES_REQUESTED** [review]

**Comment by rasom** [comment]
@alwx any updates on this PR?

**Comment by alwx** [comment]
@rasom coming today

**Comment by alwx** [comment]
Somebody made tons of indentation changes, thus this branch required rebasing. I rebased it and fixed all issues mentioned here before. 
Can be checked now.

**Comment by rasom** [comment]
> Somebody 

yeah, that was me

**Comment by rasom** [comment]
there are also formatting errors
https://jenkins.status.im/blue/organizations/jenkins/status-react%2Fpull%20requests/detail/PR-4058/8/pipeline

<blockquote></blockquote>

**Review by flexsurfer — APPROVED** [review]

**Review by rasom — APPROVED** [review]

**Comment by alwx** [comment]
Fixed.

**Comment by cammellos** [comment]
@alwx I had encountered some problems when changing schema version on develop as this code https://github.com/status-im/status-react/blob/3b8e2b7f75796813b5dea988597b3bc211f114f5/src/status_im/data_store/realm/core.cljs#L66 is triggered everytime I log in and out, resetting the database. Could you check if you have the same issue? (log in/log out, or restart the app, you loose all the chats). If so we probably want to remove that check with the PR.

**Comment by flexsurfer** [comment]
@janherich please take a look at the comment above

File: ios/StatusIm.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1566,6 +1587,13 @@
 			remoteRef = B24FC7F11DE718EF00D694FF /* PBXContainerItemProxy */;
 			sourceTree = BUILT_PRODUCTS_DIR;
 		};
+		B274C68320A069B400B55C41 /* libRNSecureRandom.a */ = {
+			isa = PBXReferenceProxy;
+			fileType = archive.ar;
+			path = libRNSecureRandom.a;
+			remoteRef = B274C68220A069B400B55C41 /* PBXContainerItemProxy */;
+			sourceTree = BUILT_PRODUCTS_DIR;
+		};
```
Comment (janherich) [inline ios/StatusIm.xcodeproj/project.pbxproj]: Why are changes in this file necessary for this PR ?

**Comment by janherich** [comment]
@flexsurfer @cammellos @alwx Yes, we should remove that check and include it conditionally whenever some future release will contain incompatible changes.
In my opinion, it should be done via flag and it should be simple check if schema version (`realm-version file-name encryption-key`) is different (`=`) from provided schema version `and` it's not `-1` (value for fresh install without any previous schema installed), if both conditions are false and flag is set, we just show the popup and delete the schema, if not, normal flow (migration to latest version) will proceed.
In case of any incompatible release in the future, it will be just matter turning the flag on (default will be false).
I can open PR with that logic, in the meantime you should remove that strange version check in any PR which increases the schema version.

**Comment by alwx** [comment]
@janherich project.pbxproj changes have been removed

**Comment by alwx** [comment]
@janherich @cammellos if I got it right, the problem should be solved by this commit (also included to this PR):
https://github.com/status-im/status-react/pull/4058/commits/4764313e35e9d3bed5a517b6d086d39fa3c82d33

**Review by janherich — APPROVED** [review]

**Review by cammellos — APPROVED** [review]

**Comment by cammellos** [comment]
@alwx code style check failed https://jenkins.status.im/blue/organizations/jenkins/status-react%2Fpull%20requests/detail/PR-4058/12/pipeline 

<blockquote></blockquote>

**Comment by yevh-berdnyk** [comment]
Android: https://i.diawi.com/qpU67y
iOS: https://i.diawi.com/1Kq1EJ

Found issues
**Transaction phrase is changed after recovering an account in both cases if wallet was set up or it was not**
Reproduction:
- On device 1 create new account
- Navigate to Wallet and remember a transaction phrase
- Setup your wallet (you can skip this step and just remember the phrase)
- On device 2 recover created account
- Click Wallet
- Click Send Transaction
- Select a recipient and input some amount
- Click Sign transaction
- Transaction phrase is changed


<blockquote><div><strong><a href="https://i.diawi.com/qpU67y">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>
<blockquote><div><strong><a href="https://i.diawi.com/1Kq1EJ">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by alwx** [comment]
This issue is not related to onboarding flow

**Comment by yevh-berdnyk** [comment]
@alwx you right, I've checked on develop and wallet phrase is changed after recovering account.
I'll create a separate issue for it

**Comment by goranjovic** [comment]
@alwx I see two commits in the PR. Is this intentional or do you want to squash them before merge?

**Comment by flexsurfer** [comment]
@alwx could you resolve conflicts and squash commits please

**Pull request merged**